com.vrcfury.vrcfury/Editor/VF/Feature/RadialToggleOptimizerBuilder.cs

```diff
@@ -0,0 +1,245 @@
+                local.TransitionsTo(sendState)
+                    .When(localPointer.IsEqualTo(i));
+            }
+
+            var mappedDict = new Dictionary<string, string>();
+            for (int i = 0; i < toOptimize.Count; i++) {
+                var dst = toOptimize[i];
+                var rxState = layer.NewState($"Receive {toOptimize[i].Name()}");
+                rxState.DrivesCopy(dst, data, false)
+                    .TransitionsToExit()
+                    .When(fx.Always());
+                remote.TransitionsTo(rxState)
+                    .When(pointer.IsEqualTo(i));
+
+                var dstSmoothed = smoothingService.Smooth($"{dst.Name()}/Smoothed", dst, 0.1f, false);
+                var dstMapped = math.SetValueWithConditions($"{dst.Name()}/Mapped",
+                    (dst, math.GreaterThan(isLocalFloat, 0.5f)),
+                    (dstSmoothed, null)
+                );
+
+                mappedDict[markedDict[dst.Name()].Name()] = dstMapped.Name();
+            }
+
+            fx.GetRaw().RewriteParameters(name => {
+                if (mappedDict.TryGetValue(name, out var mapped)) return mapped;
+                return name;
+            }, false, layers.Select(l => l.stateMachine).ToArray());
+            
+            var toCleanup = markedDict.Values.Select(prm => prm.Name()).ToHashSet();
+            for (int i = 0; i < fx.GetRaw().parameters.Length; i++) {
+                var prm = fx.GetRaw().parameters[i];
+                if (!toCleanup.Contains(prm.name)) continue;
+                fx.GetRaw().RemoveParameter(i);
+                i--;
+            }
+
+            Debug.Log($"Radial Toggle Optimizer: Reduced {toOptimize.Count * 8} bits into 16 bits.");
+        }
+
+
+        private List<VFAFloat> GetAllSyncedRadialPuppetParameters() {
+            var floatParams = new List<VFAFloat>();
+            manager.GetMenu().GetRaw().ForEachMenu(ForEachItem: (control, list) => {
+                if (control.type != VRCExpressionsMenu.Control.ControlType.RadialPuppet)
+                    return VRCExpressionsMenuExtensions.ForEachMenuItemResult.Continue;
+                var controlParam = control.GetSubParameter(0)?.name;
+                if (string.IsNullOrEmpty(controlParam))
+                    return VRCExpressionsMenuExtensions.ForEachMenuItemResult.Continue;
+                var vrcParam = manager.GetParams().GetParam(controlParam);
+                if (vrcParam == null || vrcParam.networkSynced == false)
+                    return VRCExpressionsMenuExtensions.ForEachMenuItemResult.Continue;
+                var animParam = GetFx().GetRaw().GetParam(control.GetSubParameter(0).name);
+                if (animParam != null && animParam.type == AnimatorControllerParameterType.Float)
+                    floatParams.Add(new VFAFloat(animParam));
+                return VRCExpressionsMenuExtensions.ForEachMenuItemResult.Continue;
+            });
+            return floatParams;
```

Comment:
If the same float param is used in multiple menu items then the dictionary creation will throw an error, was able to fix locally by changing to "return floatParams.GroupBy(x => x.Name()).Select(x => x.First()).ToList();" to remove duplicates

Reply:
Resolved by using a HashSet instead. https://github.com/VRCFury/VRCFury/pull/238/commits/a43fd0ea601e86cc77f02f54eecd72c19253fbfa